dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/handle/AbstractHandle.java

```diff
@@ -22,5 +22,5 @@
 public abstract class AbstractHandle implements Handle {
 
   @Override
-  public void close() throws IOException {}
+  public final void close() throws IOException {}
```

Comment:
It doesn't look like correct direction for `AbstractHandle`. This change means that any subclass of `AbstractHandle` will not be able to close any resource properly in `close` method. 

We can:
1. leave empty `close()` method.  - In this case the motivation of abstract classes is to override `close` only if needed
2. remove `AbstractHandle` completely. - In this case subclasses will explicitly override `close()`. 
3. add something like `default void close() throws IOException {}` to `Handle`. It's not very popular practice because `default` was introduced for backward compatibility with older java versions and such usage is something like `hack`, but it's still valid.

Reply:
Agree with Vladislav - this shouldn't be final, it would defeat the point.

I would keep AbstractHandle - I _always_ have an interface and an abstract class, even if it's empty, for any subclass hierarchy. One always needs it sooner or later.

Side note: I don't think of default as a hack, it's a great way of writing mixin functionality based on a small set of underlying state methods.

Reply:
I also wouldn't remove AbstractHandle. It would be a large refactor and I don't think it would improve the code.

At first my understanding of this hierarchy was that:
1. `Handle` represents resources which _may_ need to be closed
2. We've decided to always make the closing step (because a redundant `close` is better than a missing one)
3. For some `Handle`s the closing step will be trivial, for which we use `AbstractHandle`

I see now that 3. is not true, because the actual reason for AbstractHandle is to follow a more general convention about interfaces.

Reply:
It seems that during this conversation we come to the conclusion that we should leave the `AbstractHandle` without any changes. 
I'm totally ok with it.